Merge Sites

flynnduism opened the pull request:
Looking to merge the `helm.sh` Jekyll site and the `docs.helm.sh` Hugo site into a single, Hugo-based app for easier deployment and hosting.

flynnduism commented:
This PR is getting pretty big. I've gotten the refactor working for the most part, the site deploys successfully to a staging site (on Netlify) as a single app: 

https://brave-brahmagupta-4ced99.netlify.com/

---   

### Changes

* deleted the jekyll `helm.sh` migrated the content into hugo
* moved the hugo `docs.helm.sh` site to root
* various gulpfile and package changes to work in this new format
* merging sass files and assets

---    

The updated codebase is structured as follows:

```
/                               // helm.sh home
/blog/                          // blog listing
/blog/{{ post url }}            // single blog post
/docs/                          // docs homepage
/docs/{{ section }}             // docs main sections
/docs/{{ section }}/#article    // docs articles
```

---  

### To Dos

- [ ] fix paths for icon fonts and klinic slab typeface
- [ ] fix template for blog posts
- [ ] fix the docs sidebar links (use absolute urls)
- [ ] restore the docs homepage template

flynnduism commented:
Ok, this PR is ready for review. Looking for tyre kickers! 🎉

Summary of changes

- merge to single hugo site (landing page, blog, docs)
- docs still imported from [helm/helm](https://github.com/helm/helm/tree/master/docs) during build
- removed jekyll
- visual changes to nav, blog
- pipeline changes


## Reviewing

This PR is testable on Netlify here: [vigorous-banach-0c4d85.netlify.com](https://vigorous-banach-0c4d85.netlify.com/)

Changes to the site development and local dev can be found in [the updated README](https://github.com/helm/helm-www/blob/5bee49165c35575626860f14c8d92720737f93db/README.md).

## One Site, One Nav

Preview: 

![helm-nav](https://user-images.githubusercontent.com/686194/51205707-5aa19280-18bb-11e9-83a6-264246ad006f.gif)

Do we have any other top-level links we would want included across the board?

## Blog Styling

There have been a couple of issues around the legibility of the blog (currently white text on top of the blue ocean backdrop), so the style here is changed to follow the docs and offer a cleaner, high-contrast layout. 

![screen shot 2019-01-15 at 1 52 37 pm](https://user-images.githubusercontent.com/686194/51212268-df48dc80-18cc-11e9-87f8-e85f4090ba15.png)

--

flynnduism commented:
Finished some rebasing and tidying up here, so checks are passing now.

Site changes can be previewed here: [helm-merge.netlify.com](https://helm-merge.netlify.com/)

bridgetkromhout commented:
I noticed that the site preview seems to have lost the change introduced in https://github.com/helm/helm-www/pull/163. Is that a sign that we forked months ago and may be missing other content updates?

bridgetkromhout commented:
Current production site, on the front page:
<img width="317" alt="screen shot 2019-01-16 at 5 17 04 pm" src="https://user-images.githubusercontent.com/2104453/51284934-e2b69380-19b2-11e9-8bde-fe9543dd2bbf.png">

Preview site for updates:

<img width="354" alt="screen shot 2019-01-16 at 5 16 59 pm" src="https://user-images.githubusercontent.com/2104453/51284943-eba76500-19b2-11e9-8f89-5027502fa255.png">

The code in question: `<h2>Why Teams <i class="fa fa-heart"></i> Helm</h2>`

Do we think the font-awesome stuff just isn't being built in the preview but will be okay? Or is it missing?


 

bridgetkromhout commented:
```
Helm is supported by and built with a community of over 250 developers.
```

It seems oddly static to hard-code such a number in. Is this an artifact of the count of github contributors at some point in time?

bridgetkromhout commented:
I'm seeing that the content of `helm-www/helm.sh/_includes/footer.html` is what's actually being used in production. It includes this:
```
      <a target="_blank" href="https://www.samsungsds.com"><img src="{{site.baseurl}}/assets/images/logowall/samsungsds.png" alt="Samsung SDS"></a>
```

That was added December 10th: https://github.com/helm/helm-www/commit/10a60d45be954349beac2a85e27ec175d2623e21 - but the Samsung icon is missing from the version showing in test. This ties in to my previous observation: I don't have a high degree of confidence that what we're looking at in test is giving us a clear view of what we'll see in prod after a merge. I haven't investigated the disconnect further yet, but there's definitely something amiss.

bridgetkromhout commented:
More font-awesome icons missing:

<img width="403" alt="screen shot 2019-01-16 at 6 03 54 pm" src="https://user-images.githubusercontent.com/2104453/51286663-81de8980-19b9-11e9-83dd-c0c4dc074d13.png">

vs
<img width="407" alt="screen shot 2019-01-16 at 6 04 06 pm" src="https://user-images.githubusercontent.com/2104453/51286668-85721080-19b9-11e9-8c7c-ce423e216140.png">

bridgetkromhout commented:
Content issue (non-blocking for the release) - the sig-apps youtube embed is to a video from 2016 hosted on an unused-at-this-juncture deis account. Recent sig-apps youtube links seem to be on Matt Farina's personal account and there isn't a single playlist we can link.

bridgetkromhout commented on config.toml:
```diff
@@ -1,15 +1,15 @@
-baseurl = "https://docs.helm.sh/"
+baseurl = "https://helm-merge.netlify.com/"
```
Do we want to change this back pre-merge?

bridgetkromhout commented:
I'm noticing that this PR adds lots of files to `content/docs` which I think the build is pulling in from `helm/helm`. Is that desired? Should these exist in `helm/helm-www` now?

flynnduism commented:
Thanks for the super review Bridget! Lots of great catches.

I've updated with bug fixes, corrections and rolled back those content regressions.

---

## Porting Content Updates 
You're right - there were missing commits. Since this rewrite PR started in mid-dec (on 12/07, around Kubecon), there's been a bunch of content edits that I had been cross-checking manually, as site structure changes were making normal rebasing more or less impossible. It's been messy!

I've done a more thorough run through (going commit-by-commit), ensuring edits from the last 7 weeks are all included properly. 

## Icons
:facepalm: `font-awesome` was missing! I've fixed that now.

## Image Mismatch
Prior to merging, the two sites had a _similar_ set of images, but with some differences like this one. I've done a thorough review image by image, confident that any other mismatches from the merge are resolved.
Corrected images:

```
btn-get-helm.png
blog/helm-hub.png
cncf-color copy.png  (compression)
geometry3.png
phippy-sm-smiling.png
logowall/samsungsds.png
```

## 250 Developers
Yeah this is well out of date at this point; a marketing stat used to reflect community trust in the project. Now this figure is at 420 contributors... I've updated the page to reflect.

## Youtube Content
Good point. I think we should remove the stale embed and link out to youtube. Given the mixture of accounts/playlists, directing folks to a search is the simplest way to point to freshest content.
[https://www.youtube.com/results?search_query=kubernetes+sig+apps](https://www.youtube.com/results?search_query=kubernetes+sig+apps)

## Config Baseurl
Yep, this will be switched to the prod domain as part of the [launch checklist](https://github.com/helm/helm-www/pull/162#issuecomment-446382002) above

## content/docs/ Dir
Better to remove these. The `content/docs/` directory is used during the site build, for the fetching and processing of docs, but you're correct; they shouldn't be included in version control - could be interpreted as a source of docs and lead to edits in the wrong place. I've gitignored this content.

flynnduism commented:
> I'm seeing that the content of `helm-www/helm.sh/_includes/footer.html` is what's actually being used in production. It includes this:
> 
> ```
>       <a target="_blank" href="https://www.samsungsds.com"><img src="{{site.baseurl}}/assets/images/logowall/samsungsds.png" alt="Samsung SDS"></a>
> ```
> That was added December 10th: [10a60d4](https://github.com/helm/helm-www/commit/10a60d45be954349beac2a85e27ec175d2623e21) - but the Samsung icon is missing from the version showing in test. This ties in to my previous observation: I don't have a high degree of confidence that what we're looking at in test is giving us a clear view of what we'll see in prod after a merge. I haven't investigated the disconnect further yet, but there's definitely something amiss.

Thanks for the deep dive @bridgetkromhout. I've run through the feedback, made changes and added some [notes here](https://github.com/helm/helm-www/pull/162#issuecomment-455434438) (Github seems to be squashing all of those posts now :| ), thanks for taking the time to look at this!

bacongobbler commented:
I was able to get this working, however whenever I clicked on a link in the docs sidebar I'd get a 404.

For example, clicking on "Using Helm" gives me the URL http://localhost:1313/using_helm/ (a 404). The correct URL should be http://localhost:1313/docs/using_helm/. Perhaps the subpath links are relative to the root URL rather than to the subpath?

bacongobbler commented:
Hmm, upon further inspection this only seems to be an issue on certain hyperlinks:

- Using Helm
- Helm Commands
- Charts
- Developing Templates
- Best Practices

Everything else seems to be fine.

bacongobbler commented:
Everything else looks good to me! LOOOOVE the style updates to the blog site. ❤️

flynnduism commented:
Thanks for the review on this. Will take a look soon and get this ready to go.

flynnduism commented:
> A few changes required to fix the docs sidebar links. Everything else works great!

Thanks for this. I've updated the Docs sidebar urls now to correct the issue.

bacongobbler commented:
LGTM

bridgetkromhout commented:
Looks great!
